subsys/bluetooth/host/long_work.h

```diff
@@ -0,0 +1,31 @@
+/* long_work.h - Workqueue intended for long-running operations. */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#if defined(CONFIG_BT_LONG_WQ)
+
+int bt_long_work_schedule(struct k_work_delayable *dwork, k_timeout_t timeout);
+int bt_long_work_reschedule(struct k_work_delayable *dwork, k_timeout_t timeout);
+int bt_long_work_submit(struct k_work *dwork);
+
+#else
+
+static inline int bt_long_work_schedule(struct k_work_delayable *dwork, k_timeout_t timeout)
+{
+	return k_work_schedule(dwork, timeout);
+}
+
+static inline int bt_long_work_reschedule(struct k_work_delayable *dwork, k_timeout_t timeout)
+{
+	return k_work_reschedule(dwork, timeout);
+}
+
+static inline int bt_long_work_submit(struct k_work *work)
+{
+	return k_work_submit(work);
+}
+
+#endif /* CONFIG_BT_LONG_WQ */
```

Comment:
I don't recall if we ever discussed this, but I am not a huge fan of silently using the system workqueue when calling `bt_long_work_reschedule`, as the name implies a specific workqueue. 

If everyone else is OK with this approach, I guess it's fine

Reply:
I think @jhedberg had an ok argument there https://github.com/zephyrproject-rtos/zephyr/pull/46131#discussion_r889127801 (also my arg pretty much).